### DISCUSSION TITLE
How to make 24/7 hosting bot setup? [Using PM2 Module]

### DISCUSSION THREAD

**Discussion by d0biar** [Refloow/Steam-Card-Bot-PRO#16] [Q&A] [answered]
How to make 24/7 hosting bot setup? [Asked by a Discord memeber]

**Comment by Refloow** [comment, marked as answer]
**There is multiple solutions but using [PM2](https://pm2.keymetrics.io) module would be an optimal solution.**

### 1. Setup the module

1. Open Command prompt, and open directory of the project files 
2. Use command `npm install pm2 --save-prod`
command will add pm2 to module dependencies.

Another solution would be to manually add pm2 to `"dependencies"` as `"pm2": "latest",` to `package.json` 
 In case of adding pm2 manually to `package.json` 
- Watch out not to mess up the syntax.
- Run install.bat to install the module after adding module to `package.json`.


### 2. Run the script using the module

1. Open Command prompt, and open directory of the project files 
2. Use command `pm2 start index.js`

Another solution would be to manually add command `pm2 start index.js` to start.bat instead of `node index.js`
